### PR TITLE
Main patch tm v0.34.15

### PR DESCRIPTION
**Pull request by tnasu**
#### STATUS: commit hash

- SKIP: https://github.com/tendermint/tendermint/commit/31b3e279fc17d06c7a30d99114ff219bf8ce768b
- SKIP: https://github.com/tendermint/tendermint/commit/876b3c0dbe6002931b076a16c5d0820458447755
- SKIP: https://github.com/tendermint/tendermint/commit/46a6691e11c6d9b2eb3f2af4af07d0801eb25e25
- SKIP: https://github.com/tendermint/tendermint/commit/3567d3ab38ea9983f640263d783f099692b65a5b
- SKIP: https://github.com/tendermint/tendermint/commit/f721bf5154fd593606dbec91fcebf06250a118a9
- SKIP: https://github.com/tendermint/tendermint/commit/3e119fc6c41df917b6eddf914c82a3620caffb06
- PICK: https://github.com/tendermint/tendermint/commit/8a2dcbafaec0eddd960af4c7fe804a8044999c22
  - https://github.com/line/ostracon/pull/368/commits/98ef31c824a6b56facb31119487dd97da1ad7241
  - I responded to do cherry-pick since original backports aren't enough(I think it's forgotten)
  - MEMO: https://github.com/tendermint/tendermint/tree/v0.35.0-rc1/.github/linter
    - EXTRA PICK: https://github.com/tendermint/tendermint/commit/d0c87ff27e1271f79e3baeea5303d0983c40d217
      - https://github.com/line/ostracon/pull/368/commits/28832b40af0cc2626e6d2668bdf1daa95fb0b170
    - EXTRA PICK: https://github.com/tendermint/tendermint/commit/e52f9de148bed51dca424d28537e6f724d84d294
      - https://github.com/line/ostracon/pull/368/commits/1d9398f3a526bd94de58f19017c51b38de3eb7a9
    - EXTRA PICK: https://github.com/tendermint/tendermint/commit/865234e113547c34bcf6e69883e9a5c9f4c6db12
      - https://github.com/line/ostracon/pull/368/commits/9c21d6630633fcbe36cfa40b1a4edd1761eb0dbf
    - EXTRA PICK: https://github.com/tendermint/tendermint/commit/fdb7421ae8df2ed8b64adb69527e392390c1a726
      - https://github.com/line/ostracon/pull/368/commits/2c69c0623eabaaa3f449c59a269404cba16eecf1
- SKIP: https://github.com/tendermint/tendermint/commit/61831cf5eff1852099b85588b640aab9866fbae0
- PICK: https://github.com/tendermint/tendermint/commit/4ef140f6ca41e7e3397e34a16d0877a51e286013
  - https://github.com/line/ostracon/pull/368/commits/83d5134084e004bc54eb1bc92a4bd5b78151be99
  - I responded to do cherry-pick since original backports aren't enough(I think it's forgotten)
  - MEMO: https://github.com/tendermint/tendermint/blame/master/rpc/openapi/openapi.yaml
    - EXTRA PICK for fixing linter: https://github.com/tendermint/tendermint/commit/0407cd52589dfc7bc8667a70e42c50bc67ea0c76
      - https://github.com/line/ostracon/pull/368/commits/c3103c37c40047d0ad069fa127d9cc7ef9edda14
  - Fix linter: https://github.com/line/ostracon/pull/368/commits/cd4d8973887fc70872bbb5e9d5870fb1edff5cd8
- SKIP: https://github.com/tendermint/tendermint/commit/113118ec0083016eb971305f7e1434524700739f
- SKIP: https://github.com/tendermint/tendermint/commit/3e7fc468e40f92d2309b8e7330a6f6c5ed198ce2
- SKIP: https://github.com/tendermint/tendermint/commit/12ecfb03838b8496638450b8021d0035349ffcbf
- SKIP: https://github.com/tendermint/tendermint/commit/0b835bea7a381ab94d4d66cf430bea73c687ffff
- SKIP: https://github.com/tendermint/tendermint/commit/bd48acb2ca072defa3b263bf8532739226fd7f03
- SKIP: https://github.com/tendermint/tendermint/commit/d35b50b52808a101af3e8cda8615511e59e0c792
- SKIP: https://github.com/tendermint/tendermint/commit/d0031b0503c91c117c07a7b282373b7e3a31edbd
- SKIP: https://github.com/tendermint/tendermint/commit/9ec863f948c70cec3a815aa83330ce93ddcc674a
- PICK: https://github.com/tendermint/tendermint/commit/12e3419f2bd3f52f0b54828b2b01ee207cd5ff77
  - https://github.com/line/ostracon/pull/368/commits/b80e5ef191912068f7b05c26e19a1be5211a0c01
- SKIP: https://github.com/tendermint/tendermint/commit/c4834df3f328b5b6da61e75dbfd90f4982aba636
- PICK: https://github.com/tendermint/tendermint/commit/9994396e594aca0b700f49b8ef4b3eb7318f9cbe
  - https://github.com/line/ostracon/pull/368/commits/e827e8dd003bd4c2680acd255f5e0b8be3aeaed6
- PICK: https://github.com/tendermint/tendermint/commit/05340ca0693c2def1fc01332fc4a39b78c6bae8d
  - https://github.com/line/ostracon/pull/368/commits/c77d1bfded37fbd9b14ff58c2ac199c7e874f9f8
  - Fix codecov: https://github.com/line/ostracon/pull/368/commits/83614f772a5be36297e0e97cde43415bcb0e3043
- SKIP: https://github.com/tendermint/tendermint/commit/41c176ccc6a75d25631d0f891efb2e19a33329dc

#### Additional work
- Add mockrey command: https://github.com/line/ostracon/pull/368/commits/b5f22de40c8b06ce9900009f4d408b13e4d2c0dd
- Refactor filenames and improve process on github actions: https://github.com/line/ostracon/pull/368/commits/76cd8da8315011e47452abba81966ef1571fc3f2

#### Upgrade work (v0.34.15 includes a lot of upgrades lib/plugin, so I responded to it all together instead of skipping the commits)
- Upgrade lib: https://github.com/line/ostracon/pull/368/commits/100d460211e8c036dbc84dd53736bbc8ddf8ed89
  - Fix lint: https://github.com/line/ostracon/pull/368/commits/2ee2c37a7ab61b25a6e0a2327ca3141b2546df52
  - Bump github.com/gorilla/websocket from 1.4.2 to 1.5.0: https://github.com/line/ostracon/pull/368/commits/35bf1858de282b796bf87f382b5767caf59235bc
- Upgrade plugin: https://github.com/line/ostracon/pull/368/commits/8a61b27cf2ce888ef6a84caf46e2a886b705a149

##### Upgrade work include fixing potential security vulnerabilities
- github.com/opencontainers/image-spec: 1.0.2: https://github.com/line/ostracon/security/dependabot/1
- github.com/opencontainers/runc: 1.0.3: https://github.com/line/ostracon/security/dependabot/2
screenshot 2022-02-25 14 56 50

### CODE REVIEW

**Review by torao — DISMISSED** [review]

**Review by kokeshiM0chi — DISMISSED** [review]
LGTM

File: crypto/vrf/vrf_coniks.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build coniks
 // +build coniks
```
[the remark this answers is not in the document]
Comment (tnasu) [reply crypto/vrf/vrf_coniks.go:2]: Yes, it does.
1. It's best to use `go:build` for latest golang version: https://pkg.go.dev/go/build#hdr-Build_Constraints
2. This update is for Lint checking
3. But our golang is based on v1.15 and it's best to use `+build`: https://pkg.go.dev/go/build@go1.15.15#hdr-Build_Constraints

**Comment by kokeshiM0chi** [comment]
Please add to description that you are able to deal with this security problem. https://github.com/line/ostracon/security/dependabot

File: crypto/vrf/vrf_coniks.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (torao) [reply crypto/vrf/vrf_coniks.go:2]: Ok, I understood.

**Comment by tnasu** [comment]
I rebased this PR once for the [`v1.0.4`](https://github.com/line/ostracon/releases/tag/v1.0.4) release tag. Please review this again after the green light of `checks`.

**Comment by tnasu** [comment]
>Please add to description that you are able to deal with this security problem. ?>https://github.com/line/ostracon/security/dependabot

Got it

**Review by torao — APPROVED** [review]
LGTM

**Review by kokeshiM0chi — APPROVED** [review]
LGTM

**Pull request merged**